quinn/src/lib.rs

```diff
@@ -330,50 +316,63 @@ impl EndpointInner {
         incoming: mpsc::Sender<NewConnection>,
         ipv6: bool,
     ) -> Self {
+        let (sender, events) = mpsc::unbounded();
         Self {
             log,
             socket,
             inner,
             incoming,
             ipv6,
-            outgoing: None,
-            pending: FnvHashMap::default(),
-            timers: FuturesUnordered::new(),
+            outgoing: VecDeque::new(),
             buffered_incoming: VecDeque::new(),
             driver: None,
+            connecting: FnvHashMap::default(),
+            connections: FnvHashMap::default(),
+            sender,
+            events,
         }
     }
 
-    /// Wake up a blocked `Driver` task to process I/O
-    fn notify(&self) {
-        if let Some(x) = self.driver.as_ref() {
-            x.notify();
+    fn drive_connecting(&mut self) {
+        for new in self.connecting.values_mut() {
+            new.poll().unwrap();
         }
     }
```

Comment:
Could the driving of an in-progress connection be done by the future returned from `connect`, rendering the oneshot currently used there unnecessary?

Reply:
Yes, look at the new `ConnectingFuture`.